c/stcbackground.c

```diff
@@ -0,0 +1,98 @@
+#include "logging.h"
+#include "stcbackground.h"
+
+#define STC_BG_ENTRIES 101
+#define STC_BG_INTERVAL 10
+
+
+static int processInterval(STCBase *stcBase, STCModule *module, int selectStatus) {
+  STCCallbackList* moduleData = (STCCallbackList*)(module->data);
+  STCIntervalCallbackData* callbackList = moduleData->callbackList;
+
+  for (int i = 0; i < STC_BG_ENTRIES; i++) {
+    STCIntervalCallbackData* t = &callbackList[i];
+    if (t->id != -1)  /* initialise */ {
+      
+      // giving ability to disable job by making timeInterval less than zero
+      // as we return callbackdata, user can make timeInterval zero
+      if(t->timeInterval<0) continue;
+
+      t->countInterval+=STC_BG_INTERVAL;
+      if (t->countInterval <= 0) {
+        STCIntervalCallback callback = t->callback;
+        callback(moduleData->server, stcBase, module, t, t->userData);
+        t->countInterval = -t->timeInterval;
+      }
+    } else {
+      break;
+    }
+  }
+  return 0;
+};
+
+static int nextSlot(STCIntervalCallbackData* callbackList) {
+  for (int i = 0; i < STC_BG_ENTRIES; i++) {
+    if (callbackList[i].id == -1)  /* initialise */ {
+      return i;
```

Comment:
Don't we need to return `id`? If not, why is the field called `id`? Should it be `bool isInitialized`?

Reply:
`id` value is same as index location in array, which makes it id
you are right not using it for any purpose other than like boolean.
still kept it, can remove it if you want.

Reply:
Well, my point is the code shouldn't be ambiguous. If someone unfamiliar looks at it and they'll probably ask, what does id have to do with initialization? Anyway, since this is in the C file and we can change it without breaking backward compatibility. At this point, I just want this code to have the decent API and work correctly. It's up to you. 

@lchudinov how does this look to you?

Reply:
`bool isInitialized` is a field with an apt name and type. `id` looks misleading to me.

Reply:
modified